persistence-modules/core-java-persistence-4/src/test/java/com/baeldung/sql/MultipleSQLExecutionLiveTest.java

```diff
@@ -0,0 +1,96 @@
+        MultipleSQLExecution execution = new MultipleSQLExecution(connection);
+        boolean result = execution.executeMultipleStatements();
+        assertTrue(result, "The statements should execute successfully.");
+
+        try (Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery("SELECT COUNT(*) AS count FROM users WHERE name IN ('Alice', 'Bob')")) {
+            resultSet.next();
+            int count = resultSet.getInt("count");
+            assertEquals(2, count, "Two records should have been inserted.");
+        }
+    }
+
+    @Test
+    public void givenBatchProcessing_whenExecuting_thenRecordsAreInserted() throws SQLException {
+        MultipleSQLExecution execution = new MultipleSQLExecution(connection);
+        int[] updateCounts = execution.executeBatchProcessing();
+        assertEquals(2, updateCounts.length, "Batch processing should execute two statements.");
+
+        try (Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery("SELECT COUNT(*) AS count FROM users WHERE name IN ('Charlie', 'Diana')")) {
+            resultSet.next();
+            int count = resultSet.getInt("count");
+            assertEquals(2, count, "Two records should have been inserted via batch.");
+        }
+    }
+
+    @Test
+    public void givenStoredProcedure_whenCalling_thenRecordsAreInserted() throws SQLException {
+        MultipleSQLExecution execution = new MultipleSQLExecution(connection);
+        boolean result = execution.callStoredProcedure();
+        assertTrue(result, "The stored procedure should execute successfully.");
+
+        try (Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery("SELECT COUNT(*) AS count FROM users WHERE name IN ('Eve', 'Frank')")) {
+            resultSet.next();
+            int count = resultSet.getInt("count");
+            assertEquals(2, count, "Stored procedure should have inserted two records.");
+        }
+    }
+    
+    @Test
+    public void givenMultipleSelectStatements_whenExecuting_thenCorrectUsersAreFetched() throws SQLException {
+        MultipleSQLExecution execution = new MultipleSQLExecution(connection);
+        execution.executeMultipleStatements();
+
+        List<User> users = execution.executeMultipleSelectStatements();
+
+        // Here we verify the correct number of users were fetched
+        assertEquals(2, users.size(), "There should be exactly two users fetched.");
+
+        List<String> fetchedUserNames = users.stream()
+                                             .map(User::getName)
+                                             .toList();
+
+        // Here, we verify that expected users are present
+        List<String> expectedUserNames = List.of("Alice", "Bob");
+        assertTrue(fetchedUserNames.containsAll(expectedUserNames), "Fetched users should match the expected names.");
```

Comment:
For complex assertions we use AssertJ, which has a nice extract method you could use here